packages/frontend/src/components/accounts/ledger/SignInLedgerWrapper.js

```diff
@@ -106,73 +111,84 @@ export function SignInLedger(props) {
         dispatch(clearAccountState());
     };
 
-    const onClose = () => {
-        Mixpanel.track('IE-Ledger Close ledger confirmation');
+    const handleContinue = () => {
+        dispatch(redirectToApp());
+    };
+
+    const handleCancel = () => {
+        dispatch(clearSignInWithLedgerModalState());
+    };
+
+    const activeView = () => {
+        if (!signInWithLedgerStatus) {
+            return VIEWS.AUTHORIZE;
+        }
         if (signInWithLedgerStatus === LEDGER_MODAL_STATUS.CONFIRM_PUBLIC_KEY) {
-            controllerHelperApi.abort();
+            return VIEWS.SIGN_IN;
         }
         if (signInWithLedgerStatus === LEDGER_MODAL_STATUS.ENTER_ACCOUNTID) {
-            dispatch(clearSignInWithLedgerModalState());
+            return VIEWS.ENTER_ACCOUNT_ID;
+        }
+        if (signInWithLedgerStatus === LEDGER_MODAL_STATUS.CONFIRM_ACCOUNTS || signInWithLedgerStatus === LEDGER_MODAL_STATUS.SUCCESS) {
+            return VIEWS.IMPORT_ACCOUNTS;
+        }
+    };
+
+    const getCurrentViewComponent = (view) => {
```

Comment:
Now that the rendered component calls only this method within a `Container` component, what would you think about inlining this to get rid of the `switch` statement altogether? Just thinking it might be easier to parse the component if it renders these directly rather than through a function.

e.g.
```
<Container ...>
    {activeView === VIEWS.AUTHORIZE && (
        <Authorize ... />
    )}
</Container>
```

Reply:
Thinking about this a bit more, I think my suggestion doesn't make things any clearer 🤔 

What if we combine `activeView` and `getCurrentViewComponent` into one function that gets rendered as a component? Then we don't need to map to `VIEW` types and take advantage of the fact that `getCurrentViewComponent` renders a component.

```js
const LedgerAction = () => {
  if (!signWithLedgerStatus) {
    return (
      <Authorize ... />
    );
  }
  ...
  return null;
}

return (
  <Container>
    <LedgerAction />
  </Container>
);
```

Reply:
I took this implementation from `SendContainerV2.js` component - figured that it's the way we want to implement such scenarios. But I'm ok with both solutions, just not sure which one we prefare for consistency.

Reply:
IMO the two big things here are
- the unnecessary abstraction that comes with mapping application state to `VIEW` types
- returning components from invoking functions with arguments rather than React components with props

By merging the functions we get a solution with less indirection that is easier to read because it just boils down to "render X component when Y application state". As it is here and `SendContainerV2` the relationships between application state (e.g. `signWithLedgerStatus`) and component (`<Authorize />`) are unnecessarily separated, forcing developers to follow the arbitrary connection to a `VIEW` value.

Reply:
Done:
https://github.com/near/near-wallet/pull/2595/commits/dedad83c677497d6178a1f67799f883c35b12057